MVVMIOS/MovieListViewController.swift

```diff
@@ -41,34 +49,41 @@ class MovieListViewController: UITableViewController {
             activitIndicator.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor),
             activitIndicator.centerYAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerYAnchor)
         ])
+        view.addSubview(testLabel)
+        NSLayoutConstraint.activate([
+            testLabel.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor),
+            testLabel.centerYAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerYAnchor, constant: 20)
+        ])
     }
 
     func fetchData() {
         self.viewModel.resultPresentable = self
-        viewModel.fetchDataFromApi()
+        viewModel.getDataFromApi()
     }
 }
 
 extension MovieListViewController: MovieViewModelResultPresentable {
     func setupViewState(_ with: MovieViewState) {
 
-    }
-
-    func presentResult(_ result: Result<[MovieResult], Error>?) {
-
         DispatchQueue.main.async { [weak self] in
-            if let result = result {
-                switch result {
-                case .failure(let error):
-                    print(error.localizedDescription)
-                case .success(let result):
-                    DispatchQueue.main.async { [weak self] in
-                        self?.activitIndicator.stopAnimating()
-                        print(result)
-                    }
-                }
+            guard let self = self else { return }
+            switch with {
+            case .error(let movieErrorViewState):
+                self.testLabel.textColor = .brown
+                self.testLabel.text = movieErrorViewState.errorText
+            case .loading(let movieLoadingViewState):
+                self.testLabel.textColor = .red
+                self.testLabel.text = movieLoadingViewState.loadingText
+            case .loaded(let movieLoadedViewState):
+                self.activitIndicator.stopAnimating()
```

Comment:
Should probably set the animating state based off the view state if you can. Otherwise you might go from loading state with the spinner animating to an error state and forget to change the animating state